### PR TITLE
BZ-1778362: Adding how to update HTPasswd IDP with new user list

### PR DESCRIPTION
**Pull request by bergerhoffer**
* https://bugzilla.redhat.com/show_bug.cgi?id=1778362
* https://bugzilla.redhat.com/show_bug.cgi?id=1734529

Preview: https://bz-1778362--ocpdocs.netlify.com/openshift-enterprise/latest/authentication/identity_providers/configuring-htpasswd-identity-provider.html#identity-provider-htpasswd-update-users_configuring-htpasswd-identity-provider

### CODE REVIEW

**Comment by bergerhoffer** [comment]
@openshift/team-documentation For peer review please.

**Comment by bergerhoffer** [comment]
@stlaz Can you check out this new procedure to update after adding or removing users from an HTPasswd file? I pulled the steps together from various places, and they work for me, but let me know if there is a different procedure that is preferable.

Preview: https://bz-1778362--ocpdocs.netlify.com/openshift-enterprise/latest/authentication/identity_providers/configuring-htpasswd-identity-provider.html#identity-provider-htpasswd-update-users_configuring-htpasswd-identity-provider

**Review by adellape — APPROVED** [review]
:+1:

File: modules/identity-provider-htpasswd-update-users.adoc [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+----
+$ oc get secret htpass-secret -ojsonpath={.data.htpasswd} -n openshift-config | base64 -d > users.htpasswd
+----
+
+. Add or remove users from the `users.htpasswd` file.
+
+** To add a new user:
++
+----
+$ htpasswd -b users.htpasswd <username> <password>
```
Comment (stlaz) [inline modules/identity-provider-htpasswd-update-users.adoc:30]: -bB

File: modules/identity-provider-htpasswd-update-users.adoc [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+Adding password for user <username>
+----
+
+** To remove an existing user:
++
+----
+$ htpasswd -D users.htpasswd <username>
+Deleting password for user <username>
+----
```
Comment (stlaz) [inline modules/identity-provider-htpasswd-update-users.adoc:39]: why are we documenting how to work with `htpasswd`? That's what `man` is for

File: modules/identity-provider-htpasswd-update-users.adoc [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+. If you removed one or more users, you must additionally remove existing resources for each user.
+
+.. Delete the user:
++
+----
+$ oc delete user <username>
+user.user.openshift.io "<username>" deleted
+----
++
+// TODO: A user doesn't get [immediately, or at all?] kicked out of a session until you delete the user?
```
Comment (stlaz) [inline modules/identity-provider-htpasswd-update-users.adoc:56]: you need to remove all of their oauthaccesstokens

File: modules/identity-provider-htpasswd-update-users.adoc (hunk 2 above)
Comment (bergerhoffer) [reply modules/identity-provider-htpasswd-update-users.adoc:39]: @stlaz We did it in the existing docs [1], so I just continued with that in the procedure here. Would you prefer that I make this step more generic (and just say to add or remove users from the htpasswd file)?

[1] https://docs.openshift.com/container-platform/4.3/authentication/identity_providers/configuring-htpasswd-identity-provider.html#identity-provider-creating-htpasswd-file-linux_configuring-htpasswd-identity-provider

File: modules/identity-provider-htpasswd-update-users.adoc (hunk 3 above)
Comment (bergerhoffer) [reply modules/identity-provider-htpasswd-update-users.adoc:56]: @stlaz What would be the command(s) to do that?

Comment (stlaz) [reply modules/identity-provider-htpasswd-update-users.adoc:56]: as per DMs, if you've already removed the user object, the user should be able to use their accesstoken anyway

File: modules/identity-provider-htpasswd-update-users.adoc (hunk 2 above)
Comment (stlaz) [reply modules/identity-provider-htpasswd-update-users.adoc:39]: I guess the htpasswd options will hardly change, and if we already have a mention there, let's keep this, then

File: modules/identity-provider-htpasswd-update-users.adoc [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+. If you removed one or more users, you must additionally remove existing resources for each user.
+
+.. Delete the user:
++
+----
+$ oc delete user <username>
+user.user.openshift.io "<username>" deleted
+----
++
+Be sure to remove the user, otherwise the user can continue using their token as long as it has not expired.
```
Comment (bergerhoffer) [inline modules/identity-provider-htpasswd-update-users.adoc:56]: @stlaz Is this wording okay per our discussion?

**Comment by bergerhoffer** [comment]
@xingxingxia Could you please review this update? Thanks!

Preview: https://bz-1778362--ocpdocs.netlify.com/openshift-enterprise/latest/authentication/identity_providers/configuring-htpasswd-identity-provider.html#identity-provider-htpasswd-update-users_configuring-htpasswd-identity-provider

**Comment by xingxingxia** [comment]
@rhpmali is the owner of IDP configuration now, he will review this PR. I'll double review.

**Comment by xingxingxia** [comment]
LGTM

**Comment by bergerhoffer** [comment]
Thanks @rhpmali and @xingxingxia! Merging.

**Pull request merged**